### DISCUSSION TITLE
Live code editor !

### DISCUSSION THREAD

**Discussion by ddoddii** [nunocoracao/blowfish#1467] [Ideas]
Hello, @nunocoracao ! 
I've always wanted a live code editor that my blog viewers can directly right code and run the code on my blog, so I made it using [Ace coding editor](https://ace.c9.io/#nav=embedding) and [Piston API](https://github.com/engineer-man/piston)

It looks like below, and currently supports python, cpp, javascript, java.

<img width="658" alt="image" src="https://github.com/ddoddii/ddoddii.github.io/assets/95014836/341919de-0279-4907-affc-837322d26b8d">

Would it be a nice feature to make a pull request?

**Comment by maciejopalinski** [comment]
That's very cool, but it definitely should **not** be included in the theme. This is just a hugo theme, not a swiss army knife for everything. This is something that you could put as a custom partial.

**Comment by nunocoracao** [comment]
Agree with @maciejopalinski this doesn't feel like a hugo theme thing. Can you give me an example for what you would use this partial?

**Reply by ddoddii** [reply]
I'm using this live code editor in my tech blog, especially in explaining algorithm problems ! I show the result of the problem step-by-step, so readers can understand more easily. Also they can code right at my posting ! It's quite useful.

**Comment by lightmon233** [comment]
Man you gotta neeed a backend server to run that code, don't you.
